src/main/scala/com/github/unchama/seichiassist/subsystems/seichilevelupgift/usecases/GrantGiftOnSeichiLevelDiff.scala

```diff
@@ -0,0 +1,52 @@
+package com.github.unchama.seichiassist.subsystems.seichilevelupgift.usecases
+
+import cats.Applicative
+import cats.effect.Bracket.catsKleisliBracket
+import cats.effect.Sync
+import com.github.unchama.generic.Diff
+import com.github.unchama.generic.algebra.typeclasses.HasSuccessor
+import com.github.unchama.minecraft.actions.SendMinecraftMessage
+import com.github.unchama.seichiassist.subsystems.breakcount.domain.level.SeichiLevel
+import com.github.unchama.seichiassist.subsystems.seichilevelupgift.domain.{
+  Gift,
+  GiftBundleTable,
+  GrantLevelUpGift
+}
+
+object GrantGiftOnSeichiLevelDiff {
+
+  import cats.implicits._
+
+  final def grantGiftTo[F[_], Player](levelDiff: Diff[SeichiLevel], player: Player)(
+    implicit F: Applicative[F],
+    sync: Sync[F],
+    send: SendMinecraftMessage[F, Player],
+    grant: GrantLevelUpGift[F, Player]
+  ): F[Unit] = {
+    val giftBundles = HasSuccessor[SeichiLevel]
+      .leftOpenRightClosedRange(levelDiff.left, levelDiff.right)
+      .toList
+      .map { level => GiftBundleTable.bundleAt(level) }
```

Comment:
```suggestion
    val giftBundles = HasSuccessor[SeichiLevel]
      .leftOpenRightClosedRange(levelDiff.left, levelDiff.right)
      .toList
      .map { level => GiftBundleTable.bundleAt(level) }

    val giftBundle = giftBundles.fold(GiftBundle.empty)(_ combine _)
```

として、 `GiftBundle` に `traverseGifts` みたいなのを追加して `.map.toList.traverse` をここに書かないでいいようにしたいですね